daemon/api_users.go

```diff
@@ -234,36 +223,13 @@ func postUsers(c *Command, r *http.Request, user *auth.UserState) Response {
 }
 
 func removeUser(c *Command, username string, opts postUserDeleteData) Response {
-	// TODO: allow to remove user entries by email as well
-
-	// catch silly errors
-	if username == "" {
-		return BadRequest("need a username to remove")
-	}
-	// check the user is known to snapd
-	st := c.d.overlord.State()
-	st.Lock()
-	_, err := auth.UserByUsername(st, username)
-	st.Unlock()
-	if err == auth.ErrInvalidUser {
-		return BadRequest("user %q is not known", username)
-	}
-	if err != nil {
-		return InternalError(err.Error())
-	}
-
-	// first remove the system user
-	if err := osutilDelUser(username, &osutil.DelUserOptions{ExtraUsers: !release.OnClassic}); err != nil {
-		return InternalError(err.Error())
-	}
-
-	// then the UserState
-	st.Lock()
-	u, err := auth.RemoveUserByUsername(st, username)
-	st.Unlock()
-	// ErrInvalidUser means "not found" in this case
-	if err != nil && err != auth.ErrInvalidUser {
-		return InternalError(err.Error())
+	u, ue := deviceStateRemoveUser(c.d.overlord.State(), username)
```

Comment:
I would use `err` instead of `ue`, which otherwise needs to be explained (Unione Europea? ;-) ).

Reply:
nitpick: you could also combine this line and the next `if` statement into a single line.